src/index.ts

```diff
@@ -495,6 +509,38 @@ export class PaymentSocket extends EventEmitter {
       // This request succeeded to reset our retry-related variables
       this.consecutiveRejectedRequests = 0
       this.retryTimeout = STARTING_RETRY_TIMEOUT
+
+      // Adjust the chunk amount
+      if (this.maxPaymentSize.isFinite() && sourceAmount.isLessThan(this.maxPaymentSize)) {
+        this.nextMaxAmount = this.maxPaymentSize
+          .minus(sourceAmount)
+          .dividedBy(2)
+          .plus(sourceAmount)
+          .decimalPlaces(0, BigNumber.ROUND_DOWN)
+        this.debug(`setting the nextMaxAmount to ${this.nextMaxAmount} (path maximum payment size is at most: ${this.maxPaymentSize})`)
+      }
+    } else if (result.code === 'F08') {
+      // Amount too large
+
+      // Use the F08 error data to determine the path Maximum Payment Size
+      const reader = oer.Reader.from(result.unauthenticatedData)
+      try {
+        const receivedAmount = highLowToBigNumber(reader.readUInt64())
+        const maximumAmount = highLowToBigNumber(reader.readUInt64())
+
+        // If we got here without an error being thrown that means the connector
+        // did actually send those values back (thanks!)
+        const scalingFactor = maximumAmount.dividedBy(receivedAmount)
+        this.maxPaymentSize = sourceAmount.times(scalingFactor).decimalPlaces(0, BigNumber.ROUND_FLOOR)
```

Comment:
what would happen if this amount came out to zero?

Reply:
Bad things. It sends forever. Fixed in https://github.com/emschwartz/ilp-protocol-paystream/pull/17/commits/e52ad07262518600a645283e00b876d50ff32d81